net-misc/asterisk/asterisk-13.29.1.ebuild

```diff
@@ -0,0 +1,322 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools linux-info systemd
+
+MY_P="${PN}-${PV/_/-}"
+
+DESCRIPTION="Asterisk: A Modular Open Source PBX System"
+HOMEPAGE="http://www.asterisk.org/"
+SRC_URI="http://downloads.asterisk.org/pub/telephony/asterisk/releases/${MY_P}.tar.gz
+	http://downloads.uls.co.za/gentoo/asterisk/gentoo-asterisk-patchset-4.08.tar.bz2"
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+IUSE_VOICEMAIL_STORAGE="
+	+voicemail_storage_file
+	voicemail_storage_odbc
+	voicemail_storage_imap
+"
+IUSE="${IUSE_VOICEMAIL_STORAGE} alsa bluetooth calendar +caps cluster curl dahdi debug doc freetds gtalk http iconv ilbc xmpp ldap libedit libressl lua mysql newt +samples odbc osplookup oss pjproject portaudio postgres radius selinux snmp span speex srtp static statsd syslog vorbis"
+IUSE_EXPAND="VOICEMAIL_STORAGE"
+REQUIRED_USE="gtalk? ( xmpp )
+	^^ ( ${IUSE_VOICEMAIL_STORAGE/+/} )
+	voicemail_storage_odbc? ( odbc )
+"
+
+PATCHES=( "${WORKDIR}/asterisk-patchset" )
```

Comment:
You should indicate the version this patcshet was introduced in, and preferably try to get (or as much of) it upstreamed.

Reply:
Patches are inherited.  They're mostly as old as the mountains (ie, we've been carrying them for longer than I care to admit).  Most of the patches I cooked (barring 02-dahdiras-without-root and 05-dundi-null-dereference.patch) has already been merged upstream and I'll try to get these upstream too.  That obviously won't make it for any of the current versions of asterisk.

Ok, so given that we have an existing patch set, what would normally be the mechanism to indicate when the patches were introduced, and when the specific grouping was introduced?

When putting patches in ${FILESDIR} I now name then ${PV}-shortname.patch|diff and then 

With only four patches left (although I've got about 10 more that I manage out of tree that possibly doesn't even belong upstream or even in gentoo) - any reason to not include them in-tree in the above manner?

I guess I could do ${ORDER}-${PV}-shortname.patch inside the tarrball and rename the tarball to gentoo-asterisk-patchset-${PV).tar.bz2 - would this satisfy your requirements?

Reply:
Oh wait, the patchset here just points to the folder created after extracting the archive, which is versioned.  Comments still relevant.

Reply:
Ok, leave them like this for now. But when you introduce new one, let's follow the good guidelines.